### PR TITLE
Source Greenhouse: unpin CDK

### PR DESCRIPTION
**Pull request by ChristoGrab**
## What

Unpinning CDK version for source Greenhouse.

## How

- Updated poetry dependencies and manifest version
- Small change to the StreamSlicer custom component to return a StreamSlice (instead of a dict) to account for changes made to the CDK in [0.68.2](https://github.com/airbytehq/airbyte/pull/35471)

### CODE REVIEW

File: airbyte-integrations/connectors/source-greenhouse/pyproject.toml [hunk 1]
```diff
@@ -17,7 +17,7 @@ include = "source_greenhouse"
 
 [tool.poetry.dependencies]
 python = "^3.9,<3.12"
-airbyte-cdk = "==0.63.2"
+airbyte-cdk = "^0"
```
Comment (bazarnov) [inline airbyte-integrations/connectors/source-greenhouse/pyproject.toml:20]: it's better to use ">=0.63.2", but not blocking on this one.

File: airbyte-integrations/connectors/source-greenhouse/source_greenhouse/components.py [hunk 2]
```diff
@@ -109,11 +110,14 @@ def stream_slices(self) -> Iterable[StreamSlice]:
                 sync_mode=SyncMode.full_refresh, cursor_field=None, stream_slice=parent_stream_slice, stream_state=None
             ):
                 parent_state_value = parent_record.get(self.parent_key)
```
Comment (bazarnov) [inline airbyte-integrations/connectors/source-greenhouse/source_greenhouse/components.py:112]: What if there is no `parent_state_value`?

**Review by bazarnov — APPROVED** [review]

File: airbyte-integrations/connectors/source-greenhouse/source_greenhouse/components.py (hunk 2 above)
Comment (ChristoGrab) [reply airbyte-integrations/connectors/source-greenhouse/source_greenhouse/components.py]: Took another look, and now realizing the variable name is a bit misleading. This is set to the value of the current parent record's primary key, so we can assume it always exists. I've updated the variable name to `parent_primary_key` for clarity.

File: airbyte-integrations/connectors/source-greenhouse/source_greenhouse/components.py [hunk 3]
```diff
@@ -25,7 +25,8 @@ def __post_init__(self, parameters: Mapping[str, Any]):
         self._state = {}
 
     def stream_slices(self) -> Iterable[StreamSlice]:
-        yield {self.request_cursor_field: self._state.get(self.cursor_field, self.START_DATETIME)}
+        slice = StreamSlice(partition={}, cursor_slice={self.request_cursor_field: self._state.get(self.cursor_field, self.START_DATETIME)})
```
Comment (brianjlai) [inline airbyte-integrations/connectors/source-greenhouse/source_greenhouse/components.py:28]: nit: if we're only yielding one slice in the subsequent line, can we just do this in one line instead

File: airbyte-integrations/connectors/source-greenhouse/source_greenhouse/components.py [hunk 4]
```diff
@@ -108,12 +109,16 @@ def stream_slices(self) -> Iterable[StreamSlice]:
...
                 sync_mode=SyncMode.full_refresh, cursor_field=None, stream_slice=parent_stream_slice, stream_state=None
             ):
-                parent_state_value = parent_record.get(self.parent_key)
-                yield {
-                    self.stream_slice_field: parent_state_value,
-                    self.request_cursor_field: self._state.get(str(parent_state_value), {}).get(self.cursor_field, self.START_DATETIME),
+                parent_primary_key = parent_record.get(self.parent_key)
+
+                partition = {self.stream_slice_field: parent_primary_key}
+                print(self.stream_slice_field)
```
Comment (brianjlai) [inline airbyte-integrations/connectors/source-greenhouse/source_greenhouse/components.py:115]: remove extra print statement

**Review by brianjlai — APPROVED** [review]
some nits. In particular please get rid of the extra print statement before merging. But don't need to block on approval

File: airbyte-integrations/connectors/source-greenhouse/source_greenhouse/components.py (hunk 3 above)
Comment (ChristoGrab) [reply airbyte-integrations/connectors/source-greenhouse/source_greenhouse/components.py]: Done 👍

File: airbyte-integrations/connectors/source-greenhouse/source_greenhouse/components.py (hunk 4 above)
Comment (ChristoGrab) [reply airbyte-integrations/connectors/source-greenhouse/source_greenhouse/components.py]: Oh whoops! Thanks for catching, removed

**Pull request merged**